apps/namada-interface/src/slices/StakingAndGovernance/actions.ts

```diff
@@ -188,25 +191,21 @@ export const postNewBonding = createAsyncThunk<
   { state: RootState }
 >(POST_NEW_STAKING, async (change, thunkApi) => {
   const { chainId } = thunkApi.getState().settings;
-  const { rpc } = chains[chainId];
-  const rpcClient = new RpcClient(rpc);
-  const accounts = thunkApi.getState().accounts.derived[chainId];
-
-  const epoch = await rpcClient.queryEpoch();
-
-  const { hash, bytes } = await createBondingTx(
-    TxWasm.Bond,
-    change,
-    epoch,
-    chainId,
-    Object.keys(accounts)
-  );
-
-  if (hash && bytes) {
-    await rpcClient.broadcastTxSync(bytes);
-  } else {
-    throw new Error("Invalid transaction!");
-  }
+  const integration = getIntegration(chainId);
+  const signer = integration.signer() as Signer;
+  signer.submitBond({
+    source: change.owner,
+    validator: change.validatorId,
+    amount: Number(change.amount) * 1_000_000,
```

Comment:
Not a big deal, but we do have a `amountToMicro` function in `@anoma/utils` that does this conversion

Reply:
Yeah yeah :D I even used it for transparent tx.